packages/react-devtools-shared/src/hooks/parseHookNames/parseSourceAndMetadata.js

```diff
@@ -679,31 +669,43 @@ function parseSourceMapsAlternate(
         throw Error(`Expected to find HookParsedMetadata for "${locationKey}"`);
       }
 
-      const sourceMapJSON = hookSourceAndMetadata.sourceMapJSON;
-      if (sourceMapJSON != null) {
-        hookParsedMetadata.metadataConsumer = withSyncPerfMeasurements(
-          'new SourceMapMetadataConsumer(sourceMapJSON)',
-          () => new SourceMapMetadataConsumer(sourceMapJSON),
-        );
-        hookParsedMetadata.sourceMapConsumer = withSyncPerfMeasurements(
-          'new SourceMapCodecConsumer(sourceMapJSON)',
-          () => SourceMapCodecConsumer(sourceMapJSON),
-        );
+      const {runtimeSourceURL, sourceMapJSON} = hookSourceAndMetadata;
 
-        const runtimeSourceURL = hookSourceAndMetadata.runtimeSourceURL;
+      // If we've already loaded the source map info for this file,
+      // we can skip reloading it (and more importantly, re-parsing it).
+      const runtimeMetadata = runtimeURLToMetadataCache.get(runtimeSourceURL);
+      if (runtimeMetadata != null) {
```

Comment:
so this was moved from `initializeHookParsedMetadata`, right?

could you elaborate as to why moving this check here is fixing cache misses? is it because this function gets called multiple times before we initialize from cache?

Reply:
Yes. We ran `initializeHookParsedMetadata` over all of the hooks before calling `parseSourceAST` (which actually set the cached values). So we'd never cache the AST for the first batch of hooks, only for later components that happened to share the same AST value.

So I moved it to check later (in `parseSourceAST`) which covers both cases.